src/Sulu/Bundle/ContentBundle/Content/Types/SingleInternalLink.php

```diff
@@ -42,12 +42,26 @@ public function write(
     )
     {
         $value = $property->getValue();
-        if ($value !== $node->getIdentifier()) {
-            parent::write($node, $property, $userId, $webspaceKey, $languageCode,$segmentKey);
-        } else {
-            // FIXME validation and an own exception in sulu/sulu
-            throw new \Exception();
+
+        if ($value === $node->getIdentifier()) {
+            throw new \InvalidArgumentException('Internal link node cannot reference itself');
+        }
+
+        parent::write($node, $property, $userId, $webspaceKey, $languageCode,$segmentKey);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function getReferencedUuids(PropertyInterface $property)
+    {
+        $uuid = $property->getValue();
+
+        if (!$uuid) {
+            return array();
         }
+
+        return array($uuid);
```

Comment:
I would use the ternary operator because of consistency with the `InternalLinks` content type.